ccm_web/server/src/api/api.service.ts

```diff
@@ -21,4 +29,42 @@ export class APIService {
       }
     }
   }
+
+  async getCourseName (userLoginId: string, courseId: number): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.get<Course>(`courses/${courseId}`)
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
```

Comment:
`error.response.body` it is redundant to log this since it has HTML content.  `error.response.statusCode` and `error.response.statusMessage` what you want to log and send the user message

![Screen Shot 2021-06-16 at 3 44 21 PM](https://user-images.githubusercontent.com/8579775/122282769-c982db80-ceb9-11eb-91fb-7efc0849e8a3.png)

You need to have return statement in the `catch` block to return to controller the message/status_code you want to send in
`{statusCode: 422, statusMessage: 'Unprocessable Entity' }`

currently still null is return and it endup as 500 error

Event with 200 case you want to send message_object/status code 
`{statusCode: 200, statusMessage: {couseName: 'course_update'}}`

Similar strategy should be applied to put/get calls for coursenames

Reply:
Let me think about this some more. Do we really need the status code if it's `200`? 

I'm also a little confused why the body is redundant. That's where Canvas puts its specific error messages. `statusMessage` seems like a separate piece of information, probably giving a generic human-readable description for the error code, right?

Reply:
Probably not, I think it will defaults to 200. If an API call return other than 200 like 201 then I feel we should return the same response code, but in this case it 200 so no need

Reply:
In the latest commits, I decided not to include the status code in successful responses. It should already be present and correct.